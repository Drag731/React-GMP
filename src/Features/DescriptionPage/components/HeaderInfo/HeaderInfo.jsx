import React from 'react';

// import './NoFilmsFound.scss'

const HeaderInfo = ({poster_path, title, tagline, release_date, runtime, overview}) => (
    <div className="b-header-info">
        <div className="b-header-info__img">
            <img src={poster_path} className="b-header-info__img-pic" alt=""/>
        </div>
        <div className="b-header-info__about">
            <div className="b-header-info__about-title">
                {title}
            </div>
            {
                tagline &&
                <div className="b-header-info__about-tagline">
                    {tagline}
                </div>
            }
            <div className="b-header-info__about-year b-year">
                <span className="b-year__release-date">{release_date}</span>
                <span className="b-year__runtime">{runtime}</span>
            </div>
            <div>{overview}</div>
        </div>
    </div>
);

export default HeaderInfo;
