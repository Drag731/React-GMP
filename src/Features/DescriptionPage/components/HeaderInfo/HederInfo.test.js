import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HeaderInfo from './HeaderInfo.jsx'

configure({ adapter: new Adapter() });

describe('<HeaderInfo  />', () => {
    it('Renders HeaderInfo component', () => {
        const obj = {
            poster_path: 'http',
            title: 'title',
            tagline: 'tagline',
            release_date: 2001,
            runtime: 120,
            overview: 'overview'
        };

        const tree = shallow(
            <HeaderInfo
                obj={obj}
            />
        );
        expect(tree.find('.b-header-info')).toHaveLength(1);
    });
});