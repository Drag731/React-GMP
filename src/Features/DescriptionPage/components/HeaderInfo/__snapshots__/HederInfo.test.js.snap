// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderInfo  /> Snapshot one 1`] = `
<div
  className="b-header-info"
>
  <div
    className="b-header-info__img"
  >
    <img
      alt=""
      className="b-header-info__img-pic"
      src={undefined}
    />
  </div>
  <div
    className="b-header-info__about"
  >
    <div
      className="b-header-info__about-title"
    />
    <div
      className="b-header-info__about-year b-year"
    >
      <span
        className="b-year__release-date"
      />
      <span
        className="b-year__runtime"
      />
    </div>
    <div />
  </div>
</div>
`;

exports[`<HeaderInfo  /> Snapshot two 1`] = `
<div
  className="b-header-info"
>
  <div
    className="b-header-info__img"
  >
    <img
      alt=""
      className="b-header-info__img-pic"
      src={undefined}
    />
  </div>
  <div
    className="b-header-info__about"
  >
    <div
      className="b-header-info__about-title"
    />
    <div
      className="b-header-info__about-tagline"
    >
      test
    </div>
    <div
      className="b-header-info__about-year b-year"
    >
      <span
        className="b-year__release-date"
      />
      <span
        className="b-year__runtime"
      />
    </div>
    <div />
  </div>
</div>
`;
